art/defences/preprocessor/thermometer_encoding.py

```diff
@@ -146,6 +146,10 @@ def estimate_gradient(self, x: np.ndarray, grad: np.ndarray) -> np.ndarray:
         :param grad: Gradient value so far.
         :return: The gradient (estimate) of the defence.
         """
+        if self.channels_first:
+            x = np.swapaxes(x, 1, -1)
```

Comment:
How about using `permute` which preserves the ordering of H and W instead of `swapaxes`?
```suggestion
            x = x.permute(0, 2, 3, 1)
```